smolyay/smolyak.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import scipy.special
+
+
+class IndexGrid:
+    """Generate index of grid points.
+
+    Attributes
+    ----------
+    dimension : int
+        Number of independent variables.
+    exactness : int
+        Level of exactness of the approximation.
+    index_per_level : list
+        Number of indexes at each level
+    """
+
+    def __init__(self, dimension, exactness, index_per_level):
+        self.dimension = dimension
+        self.exactness = exactness
+        self.index_per_level = index_per_level
+
+        if (type(index_per_level) == tuple or
+                type(index_per_level) == np.ndarray):
+            index_per_level = list(index_per_level)
+
+        if len(index_per_level) != exactness + 1:
+            raise IndexError(
+                "index_per_level must be an array with a length of {}"
+                .format(exactness + 1))
```

Comment:
Technically, the index_per_level can describe more levels than is required for the exactness. If you only want to generate the index_per_level once and use that for multiple values of exactness, you can just generate it for the highest value for exactness and then never need to generate it again. This should only check if the index_per_level describes less than the required amount.

If the index_per_level cannot contain extra information, then a check could be used to take only the required portion in a new list.

Reply:
Addressed.